now/utils.py

```diff
@@ -246,6 +246,15 @@ def get_bucket(uri, aws_access_key_id, aws_secret_access_key, region_name):
     return bucket
 
 
+def update_tags(d):
+    dict_tags = dict()
+    for chunk in d.chunks:
+        if chunk.tags:
+            dict_tags.update(chunk.tags)
+    d.tags.update(dict_tags)
```

Comment:
Thanks for addressing the comments!
One last thing from my side:

I think we can do this
```suggestion
    for chunk in d.chunks:
        d.tags.update(d.chunks.tags)
```
but now that I think about it, what will happen if there is more than one chunk with uris? One of them will be lost, right? We would update the dict first with `{uri: first_uri}`, and then `{uri: second_uri}`, do I understand it correctly?

Also, maybe it's simpler to do this update outside the threads? We could just iterate over the docs after line 285 in `maybe_download_from_s3` and update `d.tags`

Reply:
Why does the chunk have tags?

Reply:
@florian-hoenicke I am not sure how we are using that, I kept it just in case. But I think we can remove it from being there

Reply:
> Why does the chunk have tags?

We keep the original S3 uri like that. Not sure if it's needed though.
https://github.com/jina-ai/now/blob/0b05a264ca629f9e107e4adba9f3b744000a3b2d/now/utils.py#L224

> what will happen if there is more than one chunk with uris?

Also, looking at the code, if we have more than one chunk with S3 uri one of them will get lost but I don't think that's a big deal rn. 

I think we can just iterate over the `docs` in the end of `maybe_download_from_s3` and update tags:
```python
for d in docs:
    for c in d.chunks:
        d.tags.update(c.tags)
```

Reply:
Or maybe we can even do it on this iteration
https://github.com/jina-ai/now/blob/0b05a264ca629f9e107e4adba9f3b744000a3b2d/now/utils.py#L262
so that we don't have to iterate over the docs twice